vm/src/vm.rs

```diff
@@ -309,7 +341,8 @@ impl VirtualMachine {
             repr_guards: RefCell::default(),
             state: self.state.clone(),
             initialized: self.initialized,
-        }
+        };
+        Interpreter { vm }
```

Comment:
I think we can just return the vm itself here as this is only visible to this crate and the use of it function simply calls `enter`, thus it can be replaced by `crate::vm::thread::enter_vm`

Reply:
I had it return vm just because the `Interpreter.vm` field was private, and it's the same functionality anyway. I suppose it would keep the distinction between interpreter/vm more concrete though, I'll change it